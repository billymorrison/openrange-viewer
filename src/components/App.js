import React, { useState, useEffect } from "react";
import NavBar from "./NavBar";
import MainProduct from "./MainProduct";
import productJson from "../product.json";

function App() {
  const [product, setProduct] = useState({})

  useEffect(() => {
    setProduct(productJson)
  }, [])
  
  console.log(product)

  return (
    <div className="App">
      <NavBar />
      <MainProduct product={product}></MainProduct>
    </div>
  );
}

export default App;
