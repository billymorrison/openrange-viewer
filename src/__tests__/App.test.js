import renderer from 'react-test-renderer'
import App from '../components/App';

test('renders the app', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});
