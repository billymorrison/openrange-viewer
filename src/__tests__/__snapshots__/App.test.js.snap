// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app 1`] = `
<div
  className="App"
>
  <header
    className="sc-hKgILt foIpRG"
    data-testid="navbar"
  >
    <img
      alt="get the spec logo"
      className="sc-eCssSg jOvrnQ"
      src="get-the-spec-logo.png"
    />
    <form
      className="sc-bdfBwQ hDpBk"
      data-testid="search"
    >
      <input
        className="sc-gsTCUz jpVrpN"
        id="product-search"
        name="product-search"
        onChange={[Function]}
        placeholder="Primary ID"
        type="text"
      />
      <button
        className="sc-dlfnbm jjTbmM"
        onClick={[Function]}
        type="submit"
      >
        <i
          className="fas fa-search"
        />
      </button>
    </form>
  </header>
  Hello World
</div>
`;
